File: dotnet/Operon.Sdk/Models/Workstream.cs
using System;

namespace Operon.Sdk.Models;

/// <summary>
/// Workstream configuration details.
/// </summary>
public sealed class Workstream
{
    public string Id { get; set; } = string.Empty;
    public DateTimeOffset? CreatedAt { get; set; }
    public DateTimeOffset? UpdatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public string? UpdatedBy { get; set; }
    public int? Version { get; set; }
    public string? CustomerId { get; set; }
    public string? WorkspaceId { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public string? Mode { get; set; }
    public string? Type { get; set; }
    public string? Status { get; set; }
    public WorkstreamState[]? States { get; set; }
    public string? DefaultStateId { get; set; }
    public string[]? InteractionIds { get; set; }
    public string? HcsTestTopicId { get; set; }
    public string? HcsLiveTopicId { get; set; }
}

File: java/src/main/java/cloud/operon/sdk/Workstream.java
package cloud.operon.sdk;

import java.time.Instant;
import java.util.List;

/**
 * Workstream configuration entity returned by Operon APIs.
 */
public record Workstream(
    String id,
    Instant createdAt,
    Instant updatedAt,
    String createdBy,
    String updatedBy,
    Integer version,
    String customerId,
    String workspaceId,
    String name,
    String description,
    String mode,
    String type,
    String status,
    List<WorkstreamState> states,
    String defaultStateId,
    List<String> interactionIds,
    String hcsTestTopicId,
    String hcsLiveTopicId
) {
}
